packages/databricks-vscode/src/utils/onErrorDecorator.ts

```diff
@@ -0,0 +1,75 @@
+import {window} from "vscode";
+import {
+    getContextParamIndex,
+    logging,
+    Context,
+} from "@databricks/databricks-sdk";
+import {Loggers} from "../logger";
+
+interface Props {
+    popup?:
+        | {
+              prefix?: string;
+          }
+        | true;
+    log?:
+        | {
+              prefix?: string;
+          }
+        | true;
+}
+export function onError(props: Props) {
+    return function showErrorDecorator(
+        target: any,
+        propertyKey: string,
+        descriptor: PropertyDescriptor
+    ) {
+        let contextParamIndex: number = -1;
+        // Find the @context if it exists. We want to use it for logging whenever possible.
+        if (props.log !== undefined) {
+            try {
+                logging.withLogContext(Loggers.Extension)(
```

Comment:
Seeing go-like contexts in js/ts is weird. Do we ever use them for something other than loggers? Just a random question :-)

Reply:
We use them for passing down cancellation tokens to sdk operations. We had much bigger plans for these, but never went anywhere.